src/Symfony/Component/AstGenerator/AstGeneratorChain.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * Generator delegating the generation to a chain of generators.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class AstGeneratorChain implements AstGeneratorInterface
```

Comment:
For this class and `UniqueVariableScope` i agree it would make sense even if there aren't that much final classes in symfony.
But for the routing generators i would like to allow the user to easily hook into them and it could pass by inheritance.